packages/path-compare/src/index.js

```diff
@@ -50,15 +40,10 @@ const cumulativeLength = (length, value) => {
  * @type {CompareFn<string[]|undefined>}
  */
 export const pathCompare = (a, b) => {
-  // Undefined is not preferred
-  if (a === undefined && b === undefined) {
-    return 0;
-  }
-  if (a === undefined) {
-    return 1;
-  }
-  if (b === undefined) {
-    return -1;
+  // Undefined compares greater than anything else.
+  if (a === undefined || b === undefined) {
+    // eslint-disable-next-line no-nested-ternary
+    return a === b ? 0 : a === undefined ? 1 : -1;
```

Comment:
[This rule](https://eslint.org/docs/latest/rules/no-nested-ternary) exists because:

> Nesting ternary expressions can make code more difficult to understand.

It's hard for me to disagree with that.  

If we're going to disable it on a one-off basis (and again on L21 above), we should have a justification for doing so.  Otherwise, it may be better to leave it for now, then  propose we disable the rule because we feel it isn't buying us anything.

Ideally, I'd like a rule that demands an extra comment for each one-off disabled rule (much like [ban-ts-comment](https://typescript-eslint.io/rules/ban-ts-comment/) configured with `'ts-expect-error': 'allow-with-description'`). 😄  I bet someone solved this already...

Reply:
What I think we really want is a maximum ternary nesting depth of 1 rather than 0, to allow for precisely this kind of use in comparison functions (`$equal ? 0 : $aLessThanB ? -1 : 1`).